flink-filesystems/flink-hadoop-fs/src/main/java/org/apache/flink/runtime/util/HadoopUtils.java

```diff
@@ -247,4 +248,43 @@ private static boolean addHadoopConfIfFound(
         }
         return foundHadoopConfiguration;
     }
+
+    /**
+     * Set up the caller context [[callerContext]] by invoking Hadoop CallerContext API of
+     * [[org.apache.hadoop.ipc.CallerContext]], which was added in hadoop 2.8.
+     */
+    public static void setCallerContext(String callerContext,
+                                        org.apache.flink.configuration.Configuration flinkConfig) {
+        if (isMinHadoopVersion(2,8)) {
+            try {
+                callerContext = truncateCallerContext(callerContext, flinkConfig);
+                Class callerContextClass = Class.forName("org.apache.hadoop.ipc.CallerContext");
+                Class builder = Class.forName("org.apache.hadoop.ipc.CallerContext$Builder");
+                Constructor builderInst =  builder.getConstructor(callerContext.getClass());
+                callerContextClass.getMethod("setCurrent", callerContextClass)
+                        .invoke(null, builder.getMethod("build")
+                                .invoke(builderInst.newInstance(callerContext)));
```

Comment:
We could avoid reflection if we bump Hadoop min version to 2.8. But that's probably a bigger effort that should be independent. If you want to take that as a prerequisite, I could create a new ticket.

Reply:
I know it is a graceful way to do this feature. But as you see, that's probably a big effort for this feature. Above all, I think reflection is a transitional way to do this work. And I can do the Hadoop version bumping after this feature, it may take some time, but i think i can take this ticket.